codex/blocktype.nim

```diff
@@ -154,17 +109,58 @@ func new*(
     verify: bool = true
 ): ?!Block =
   ## creates a new block for both storage and network IO
+  ##
+
+  if verify:
+    let
+      mhash = ? cid.mhash.mapFailure
+      computedMhash = ? MultiHash.digest($mhash.mcodec, data).mapFailure
+      computedCid = ? Cid.init(cid.cidver, cid.mcodec, computedMhash).mapFailure
+    if computedCid != cid:
+      return "Cid doesn't match the data".failure
+
+  return Block(
+    cid: cid,
+    data: @data
+  ).success
+
+proc emptyCid*(version: CidVersion, hcodec: MultiCodec, dcodec: MultiCodec): ?!Cid =
+  ## Returns cid representing empty content, given cid version, hash codec and data codec
   ## 
 
-  let
-    mhash = ? cid.mhash.mapFailure
-    b = ? Block.new(
-      data = @data,
-      version = cid.cidver,
-      codec = cid.mcodec,
-      mcodec = mhash.mcodec)
+  const
+    Sha256 = multiCodec("sha2-256")
+    Raw = multiCodec("raw")
+    DagPB = multiCodec("dag-pb")
+    DagJson = multiCodec("dag-json")
+
+  var index {.global, threadvar.}: Table[(CIDv0, Sha256, DagPB), Result[Cid, CidError]]
```

Comment:
The correct type here is `MultiCodec`, the reason this kinda works with this declaration, is because `DagPB` is an alias to `MultiCodec`, but this declaration is weird, because you then use `DagJson` and `Raw`, which are clearly not `DagPB`. I would declare this as `Table[(CidVersion, MultiCodec, MultiCodec), Cid]` (without the encapsulated result, so that we don't have to unpack it every time).

Reply:
Right, sorry I was rushing and my brain for some reason put wrong types here, the correct types are `(CidVersion, MultiCodec, MultiCodec)` as you pointed.

I'll also fix the value type to just `Cid` instead of `Result[Cid, CidError]`.